### PR TITLE
Data prep on multiple GPUs

### PR DESCRIPTION
**Pull request by eitanturok**
Fixes `Bug in convert_dataset_hf` #1575.

Running
```
python scripts/data_prep/convert_dataset_hf.py --dataset allenai/c4 --data_subset en --out_root ./my-copy-c4 --splits train_small val_small --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b --eos_text '<|endoftext|>'
```
now works on multiple GPUs.

### CODE REVIEW

**Comment by eitanturok** [comment]
Not sure if this is the best place to set the world_size. LMK if you have suggestions of a better place to put this.

**Comment by snarayan21** [comment]
@eitanturok is this change needed for the rest of the dataset scripts as well? they probably call build_tokenizer too, mind testing those if possible?

**Review by dakinggg — COMMENTED** [review]
can you do this for all the data prep scripts?

**Review by snarayan21 — COMMENTED** [review]
@eitanturok is this change needed for the rest of the dataset scripts as well? they probably call build_tokenizer too, mind testing those if possible?

File: scripts/data_prep/convert_dataset_hf.py [hunk 1]
```diff
@@ -48,6 +49,8 @@ def parse_args() -> Namespace:
 
 if __name__ == '__main__':
     args = parse_args()
+    # set `WORLD_SIZE` to fix https://github.com/mosaicml/llm-foundry/issues/1575
```
Comment (dakinggg) [inline scripts/data_prep/convert_dataset_hf.py:52]: i think better in `convert_dataset_hf_from_args`?

**Comment by Extirpater** [comment]
When I try to run composer -n 1 ./data_prep/convert_dataset_hf.py --dataset bigcode/the-stack-dedup --out_root s3:/XXX--splits train --concat_tokens 4096 --tokenizer XXX --eos_text '<|EOS|>' 

none of the gpus are being utilized. Is there anything that needs to be set other than world size

**Comment by KuuCi** [comment]
Running on:
image


image

image

image

image

image

**Review by snarayan21 — APPROVED** [review]
@KuuCi thanks for taking this one over!

changes seem fine to me and ty for testing.

**Comment by snarayan21** [comment]
@Extirpater These data prep scripts don't use GPU so not seeing GPU utilization when running these scripts is expected.

**Review by mvpatel2000 — APPROVED** [review]
ty!

**Pull request merged**